includes/class-wc-product-factory.php

```diff
@@ -45,6 +45,9 @@ public function get_product( $product_id = false, $deprecated = array() ) {
 		try {
 			return new $classname( $product_id, $deprecated );
 		} catch ( Exception $e ) {
+			if ( ! defined( 'REST_REQUEST' ) || ! REST_REQUEST ) {
```

Comment:
This makes it not output the error message during API requests. They'll break the json if present.